internal/pkg/service/common/etcdop/watch.go

```diff
@@ -228,6 +227,13 @@ func (v Prefix) Watch(ctx context.Context, client etcd.Watcher, opts ...etcd.OpO
 			// Pass the response
 			outCh <- resp
 		}
+
+		// Send init error, if the context has been cancelled before the "Created" event
+		if init {
+			resp := WatchResponse{}
+			resp.InitErr = ctx.Err()
+			outCh <- resp
+		}
```

Comment:
V testoch sa stalo, ze `ctx` skoncil skor, ako sa stihol vytvorit watcher, a teda ak kod cakal na `Created/InitErr` event, tak sa nedockal  a nastal timeout.

Reply:
Je to za `for` loop, ktora sa ukonci, ked sa ukonci `ch`, a ten sa ukonci ak sa ukonci `ctx`.